.github/workflows/build-user-config.yml

```diff
@@ -161,3 +176,25 @@ jobs:
         with:
           name: ${{ inputs.archive_name }}
           path: ${{ env.build_dir }}/artifacts
+
+  publish:
+    runs-on: ubuntu-latest
+    if: ${{ inputs.release_enable && !cancelled() }}
+    needs: build
+    name: Publish firmware
+    steps:
+      - name: Delete previous release
```

Comment:
Why would you want to delete previous releases? As noted above, I don't think we want to put in any behaviors that rewrite Git history, especially not by default.